### DISCUSSION TITLE
Parseq generation stops in the middle when using TensorRT

### DISCUSSION THREAD

**Discussion by Zafflower** [rewbs/sd-parseq#231] [Q&A] [answered]
Hi, I've been using Parseq together with TensorRt for a while and I've noticed a problem.  Parseq will always finish a generation when TensorRt is turned off and will sometimes finish a generation when TensorRt is turned on.  But quite often, the generation will quit in the middle when using TensorRt.  This seems to happen most when I'm using Fade Frames at the point where a new prompt starts, or when the the positive and negative prompts reach a certain level of complexity.  Please see the attached JPGs for the error message -- the message doesn't really make sense because up until the moment of failure, I am using the installed TensorRT engine that the error message wants me to install! I love Parseq, but I love TensorRT too.  My genertions take about 1/3 of the time when I use TensorRT.  I am using current versions of Stable Diffusion and the TensorRT extension.  Is there a fix for this?  Thanks.
Zaffer
![Deforum Parsec error message](https://github.com/rewbs/sd-parseq/assets/135188868/fc4d4c79-ce61-4dc2-ab5c-6da2a876ec81)

![Parseq overlapping prompts](https://github.com/rewbs/sd-parseq/assets/135188868/c1ae1ee0-bebc-4920-b451-106eac319f32)

**Comment by rewbs** [comment, marked as answer]
Hi! The problem is your tensorrt compiled model had a maximum prompt token length. By default it'll be 75 or 150. At some point during the generation you hit a frame where the full prompt exceeds that max. So you can work around this by shortening your prompts or not having composable diffusion overlaps.

The best I could do I think is add functionality that shows you the maximum token length across all frames, so you can confirm it's below your TRT max before you generate.

**Reply by Zafflower** [reply]
Thanks for the thorough answer rewbs!  When TensorRt kept dying in different places with different animations, I couldn't figure out what was happening.    It would be great if you could add the functionality that you mention.  Thanks so much for working so hard on Parseq so the rest of us can be creative with it.
Zaffer
